lib/helper.js

```diff
@@ -5,6 +5,21 @@ const hoek = require('@hapi/hoek');
 const dayjs = require('dayjs');
 const TEMPLATE_NAME_REGEX_WITH_NAMESPACE = schema.config.regex.FULL_TEMPLATE_NAME_WITH_NAMESPACE;
 const formatDate = dateTime => dayjs(dateTime).format('YYYY-MM-DD');
+// SCM handles only SUCCESS, PENDING & FAILURE status
+const SCM_STATE_MAP = {
```

Comment:
is this map kept in models and each scm plugin?

Reply:
Yes, it is handled in each scm plugin(github & gitlab) individually.

Reply:
If that is the case, is there any reason why we need it here then?

Reply:
Mappings:

In Models:
Build Status to Generic SCM Statuses(SUCCESS, PENDING, PENDING)

In Individual SCM plugins:
Generic SCM Statuses(SUCCESS, PENDING, FAILURE) to Specific SCM Statuses(e.g. FAILURE => failed in GitLab)

Reply:
If these are *build statuses* repeating them all over is only going to be confusing.

Reply:
Build statuses are repeated all over in existing system and creating confusion. That's the problem we are solving with this change.

We're defining SCM status and handling them differently with proper mappings.

In earlier versions Build Statuses were confused with SCM Statuses and code were written to handle all the build statuses in SCM, but we only need to handle 3 SCM status (SUCCESS, PENDING, FAILURE) and that we're doing with these mappings.

Please refer to @jithine's comment here on it:
https://github.com/screwdriver-cd/data-schema/pull/450#discussion_r656401198

Reply:
We need mapping from build status to inidividual SCM status check API values., we also need mapping form meta status check values to individual SCM status check API values.

What's happening with this changes is a mapping from Build status -> meta status check values (handled in models) -> individual check API status values (handled in SCM plugins)

Reply:
@ombharatiya maybe moving this to data scheme would be a better home for this. https://github.com/screwdriver-cd/data-schema/blob/master/plugins/scm.js